enhancements/monitoring/alert-overrides.md

```diff
@@ -0,0 +1,608 @@
+	// considered pending. This is represented as a Prometheus duration, for
+	// details on the format see:
+	// - https://prometheus.io/docs/prometheus/latest/configuration/configuration/#duration
+	//
+	// +kubebuilder:validation:Pattern:="((([0-9]+)y)?(([0-9]+)w)?(([0-9]+)d)?(([0-9]+)h)?(([0-9]+)m)?(([0-9]+)s)?(([0-9]+)ms)?|0)"
+	// +optional
+	For string `json:"for,omitempty"`
+
+	// labels to add or overwrite for each alert.
+	//
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// annotations to add to each alert.
+	//
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// AlertingRuleStatus is the status of an AlertingRule resource.
+type AlertingRuleStatus struct {
+	// observedGeneration is the last generation change you've dealt with.
+	//
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// prometheusRule is the generated PrometheusRule for this AlertingRule.
+	//
+	// +optional
+	PrometheusRule string `json:"prometheusRule,omitempty"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+
+// AlertRelabelConfig defines a set of relabel configs for alerts.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +k8s:openapi-gen=true
+type AlertRelabelConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec describes the desired state of this AlertRelabelConfig object.
+	Spec AlertRelabelConfigSpec `json:"spec"`
+
+	// status describes the current state of this AlertRelabelConfig object.
+	//
+	// +optional
+	Status AlertRelabelConfigStatus `json:"status,omitempty"`
+}
+
+// AlertRelabelConfigsSpec is the desired state of an AlertRelabelConfig resource.
+//
+// +k8s:openapi-gen=true
+type AlertRelabelConfigSpec struct {
+	// configs is a list of sequentially evaluated alert relabel configs.
+	//
+	// +kubebuilder:validation:MinItems:=1
+	Configs []RelabelConfig `json:"configs"`
+}
+
+// AlertRelabelConfigStatus is the status of an AlertRelabelConfig resource.
+type AlertRelabelConfigStatus struct {
+	// conditions contains details on the state of the AlertRelabelConfig, may be
+	// empty.
+	//
+	// +optional
+	Conditions []AlertRelabelConfigCondition `json:"conditions,omitempty"`
+}
+
+// AlertRelabelConfigConditionType is a valid value for the type field of an
+// AlertRelabelConfigCondition.
+type AlertRelabelConfigConditionType string
+
+const (
+	// AlertRelabelConfigReady is the condition type indicating readiness.
+	AlertRelabelConfigReady AlertRelabelConfigConditionType = "Ready"
+)
+
+// AlertRelabelConfigCondition details a status condition of an AlertRelabelConfig.
+type AlertRelabelConfigCondition struct {
```

Comment:
I expect you can just use a `metav1.Condition` rather than creating your own copy of it

Reply:
Done.